### PR TITLE
Fix development issues and poetry install (specifically related to arm64/Apply Silicon) - Drop Python 3.6 support"

### PR DESCRIPTION
**Pull request by gatli**
# **This PR drops Python 3.6 support it is end of life for security fixes for more than a year**

This PR fixes `poetry install` problems for later python versions, allowing development on all modern Python versions.
- Opens up dev requirements
- New lockfile for faster local dev 
- Drop flake8 (it was being difficult 🤯 🔫 ) - use ruff https://github.com/astral-sh/ruff

This basically required digging into different elements of the dependency tree and opening it up (which is a good idea for libraries)
-  poetry show typed-ast --tree --why

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ repos:
     hooks:
     - id: system
       name: flake8
```
Comment (ntamas92) [inline .pre-commit-config.yaml:14]: Better yet, maybe call it `lint`?

```suggestion
      name: ruff
```

**Review by ntamas92 — APPROVED** [review]
LGTM

**Review by jean-lucas — APPROVED** [review]
lgtm, tried it out on a few sample scripts locally too

File: .circleci/config.yml [hunk 2]
```diff
@@ -37,11 +37,11 @@ jobs:
       - run:
           name: Black Formatting Check # Only validation, without re-formatting
           command: |
-            poetry run black --check -t py36 .
+            poetry run black --check .
       - run:
-          name: Flake8 Lint Check # Uses setup.cfg for configuration
+          name: Ruff Lint Check # See pyproject.tooml [tool.ruff]
```
[the remark this answers is not in the document]
Comment (gatli) [reply .circleci/config.yml:42]: Oops, had automerge on 🙂

File: .pre-commit-config.yaml (hunk 1 above)
Comment (gatli) [reply .pre-commit-config.yaml:14]: Again, automerge 🙈

File: .circleci/config.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ntamas92) [reply .circleci/config.yml:42]: No worries, I'll handle these in my PR soon.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ntamas92) [reply .pre-commit-config.yaml:14]: Same

**Pull request merged**